### PR TITLE
chore(api): rename implementation classes to core

### PR DESCRIPTION
**Pull request by TamarZanzouri**
# Overview

closes https://opentrons.atlassian.net/browse/RCORE-79.
rename legacy implementation classes to core.

# Test Plan

- Test against an OT-2 with core ff off.
- Upload a python protocols to test against PAPI (this PR needs heavy smoke tests on PAPI). 

```
from opentrons import protocol_api, types

metadata = {
    "apiLevel": "2.12",
}

def run(ctx: protocol_api.ProtocolContext) -> None:
    tip_rack = ctx.load_labware("opentrons_96_tiprack_300ul", 5)
    pipette = ctx.load_instrument("p300_single", types.Mount.RIGHT, [tip_rack])
    well = tip_rack.wells()[0]
    
    # offset the well to something obvious
    # to observe that the offset is not respected
    location = types.Location(point=types.Point(x=9, y=-9, z=0), labware=well)

    pipette.pick_up_tip(location, presses=0)
    pipette.drop_tip()
```

```
metadata = {
    'protocolName': 'Heater-shaker test protocol w/ blocking commands',
    'author': 'Opentrons <protocols@opentrons.com>',
    'apiLevel': '2.13'
}

def run(context):
	# Load Heater-shaker module
	hs_mod = context.load_module("heaterShakerModuleV1", "1")

	# Load adapter + labware on module. Available labware:
	# - opentrons_96_flat_bottom_adapter_nest_wellplate_200ul_flat
	# - opentrons_96_pcr_plate_adapter_nest_wellplate_100ul_pcr_full_skirt
	# - opentrons_96_deepwell_adapter_nest_wellplate_2ml_deep
	# - opentrons_flat_plate_adapter_corning_384_wellplate_112ul_flat
	hs_plate = hs_mod.load_labware("opentrons_96_flat_bottom_adapter_nest_wellplate_200ul_flat")

	# tiprack_300ul = [context.load_labware('opentrons_96_tiprack_300ul', 3)]
	# p20 = context.load_instrument('p300_single', 'right', tip_racks=tiprack_300ul)

	tiprack_10ul = [context.load_labware('opentrons_96_tiprack_300ul', 5)]
	p20 = context.load_instrument('p300_single_gen2', 'right', tip_racks=tiprack_10ul)

	reservoir = context.load_labware('agilent_1_reservoir_290ml', 4)

	hs_mod.open_labware_latch()
	
	hs_mod.set_and_wait_for_temperature(celsius=80)  # Waits until H/S is at 80C

	p20.pick_up_tip()
	p20.aspirate(10, reservoir['A1'])
	p20.dispense(10, hs_plate['A1'])
	p20.drop_tip()

	hs_mod.close_labware_latch()
	hs_mod.set_and_wait_for_shake_speed(rpm=500)	# Waits until H/S has reached 500rpm speed

	context.delay(minutes=1)

	hs_mod.deactivate_shaker()
	hs_mod.deactivate_heater()
```

```
import time

metadata = {
    'protocolName': 'Heater-shaker test protocol w/ non blocking commands',
    'author': 'Opentrons <protocols@opentrons.com>',
    'apiLevel': '2.13'
}

SHAKE_TIME_MINUTES = 1


def run(context):
	# Load Heater-shaker module
	hs_mod = context.load_module("heaterShakerModuleV1", 1)

	# Load adapter + labware on module. Available labware:
	# - opentrons_96_flat_bottom_adapter_nest_wellplate_200ul_flat
	# - opentrons_96_pcr_plate_adapter_nest_wellplate_100ul_pcr_full_skirt
	# - opentrons_96_deepwell_adapter_nest_wellplate_2ml_deep
	# - opentrons_flat_plate_adapter_corning_384_wellplate_112ul_flat
	hs_plate = hs_mod.load_labware("opentrons_96_flat_bottom_adapter_nest_wellplate_200ul_flat")
	other_plate = context.load_labware("biorad_96_wellplate_200ul_pcr", 6)

	tiprack_10ul = [context.load_labware('opentrons_96_tiprack_300ul', 5)]
	p20 = context.load_instrument('p300_single_gen2', 'right', tip_racks=tiprack_10ul)

	reservoir = context.load_labware('agilent_1_reservoir_290ml', 4)


	hs_mod.set_target_temperature(celsius=40)  # Sets target and moves on. Doesn't wait

	p20.pick_up_tip()
	p20.aspirate(10, reservoir['A1'])
	p20.dispense(10, hs_plate['A1'])
	p20.drop_tip()

	hs_mod.wait_for_temperature()					# Cannot specify a temperature. Waits for the previously set target temperature only

	saved_time = time.time()	# saves seconds since epoch (January 1, 1970, 00:00:00)

	hs_mod.close_labware_latch()
	hs_mod.set_and_wait_for_shake_speed(rpm=500)	# Waits until H/S has reached 500rpm speed

	p20.pick_up_tip()
	p20.aspirate(10, reservoir['A1'])
	p20.dispense(10, other_plate['A1'])
	p20.drop_tip()

	context.comment(msg=f"Waiting until shaken for {SHAKE_TIME_MINUTES}min.")

	if not context.is_simulating():
		while time.time() - saved_time < SHAKE_TIME_MINUTES * 60:		# Wait until SHAKE_TIME_MINUTES since saved_time
			time.sleep(1)

	hs_mod.deactivate_shaker()
	hs_mod.deactivate_heater()
```

```
from typing import cast
from opentrons import protocol_api, types

metadata = {
    "protocolName": "Mad Mag",
    "author": "Opentrons <engineering@opentrons.com>",
    "description": "A Magnetic Module Test",
    "source": "Opentrons Repository",
    "apiLevel": "2.10",
}

def run(ctx: protocol_api.ProtocolContext) -> None:
    mm = cast(protocol_api.TemperatureModuleContext, ctx.load_module("temperatureModuleV2", 3))
    labware = mm.load_labware("nest_96_wellplate_100ul_pcr_full_skirt")
    ctx.comment(f"labware parent is {labware.parent}")
    mm.set_temperature(37)

    print(mm.status)
```

- make sure logic works as expected. there should be no change. 

# Changelog

- Changed `core/protocol_api` directory name to `core/legacy`
- Changed `core/simulator` directory name to `core/legacy_simulator`
- Changed class names from xxImplementation to LegacyxxCore
- Fixed imports and comments.

# Review requests

- Naming makes senes?
- Did I miss anything? 
- Any protocols in mind I can test with? 

# Risk assessment

Medium. Need to run smoke tests on an OT-2 with core ff off to make sure everything is working as expected.

### CODE REVIEW

File: api/src/opentrons/protocols/duration/estimator.py [hunk 1]
```diff
@@ -513,7 +513,7 @@ def z_time(is_module: bool, gantry_speed: float) -> float:
         z_default_module_height = 95.25
         # 177.8 - 82.55 Where did we get 177.8 from?
         # Would it be better to just use
-        # protocol_api.labware.Well.top
+        # legacy.labware.Well.top
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocols/duration/estimator.py]: I should probably revert this change? feels like its related to the public context

File: api/tests/opentrons/protocol_api/core/engine/test_well_core.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def well_definition() -> WellDefinition:
 def subject(
     decoy: Decoy, mock_engine_client: EngineClient, well_definition: WellDefinition
 ) -> WellCore:
-    """Get a WellCore test subject with mocked dependencies."""
+    """Get a LegacyWellCore test subject with mocked dependencies."""
```
Comment (TamarZanzouri) [inline api/tests/opentrons/protocol_api/core/engine/test_well_core.py]:
```suggestion
    """Get a WellCore test subject with mocked dependencies."""
```

File: api/src/opentrons/protocol_api/core/legacy/__init__.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+"""The Protocol API v2 context implementation package."""
+# TODO(mc, 2022-08.22): uncomment when import cycles can be resolved
+# from .protocol_context import LegacyProtocolCore
+# from .instrument_context import LegacyInstrumentCore
+# from .labware import LegacyLabwareCore
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api/core/legacy/__init__.py]:
```suggestion
# from .legacy_protocol_core import LegacyProtocolCore
# from .legacy_instrument_core import LegacyInstrumentCore
# from .legacy_labware_core.py import LegacyLabwareCore
```

File: api/src/opentrons/protocol_api/core/legacy/deck.py [hunk 4]
```diff
@@ -62,9 +62,9 @@ def _load_fixtures(self) -> None:
         for f in self._definition["locations"]["fixtures"]:
             slot_name = self._check_name(f["slot"])  # type: ignore
             # TODO(mc, 2022-06-15): this loads the fixed trash as an instance of
-            # `opentrons.protocol_api.labware.Labware`
+            # `opentrons.legacy.labware.Labware`
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api/core/legacy/deck.py]:
```suggestion
            # `opentrons.protocol_api.labware.Labware`
```

File: api/src/opentrons/protocol_api/core/legacy/deck.py [hunk 5]
```diff
@@ -62,9 +62,9 @@ def _load_fixtures(self) -> None:
...
             # However, all other labware will be added to the `Deck` as instances of
-            # `opentrons.protocol_api.core.labware.AbstractLabware`
+            # `opentrons.legacy.core.labware.AbstractLabware`
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api/core/legacy/deck.py]:
```suggestion
            # `opentrons.protocol_api.core.labware.AbstractLabware`
```

File: api/src/opentrons/protocol_api_experimental/__init__.py [hunk 6]
```diff
@@ -14,7 +14,7 @@
 
 # todo(mm, 2021-04-09): The APIv2 analogue to this file exposes some, but not
 # all, package members at the top level. For example, you can access
-# protocol_api.InstrumentContext, but not protocol_api.Point. We should take
+# legacy.InstrumentContext, but not legacy.Point. We should take
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api_experimental/__init__.py]:
```suggestion
# protocol_api.InstrumentContext, but not protocol_api.Point. We should take
```

File: api/src/opentrons/protocol_api_experimental/module_contexts/magnetic_module_context.py [hunk 7]
```diff
@@ -28,7 +28,7 @@ class MagneticModuleStatus(str, Enum):
 
 class MagneticModuleContext:  # noqa: D101
     # TODO(mc, 2022-02-09): copy or rewrite docstrings from
-    # src/opentrons/protocol_api/module_contexts.py
+    # src/opentrons/legacy/module_contexts.py
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api_experimental/module_contexts/magnetic_module_context.py]:
```suggestion
    # src/opentrons/protocol_api/module_contexts.py
```

File: api/src/opentrons/protocol_api_experimental/module_contexts/temperature_module_context.py [hunk 8]
```diff
@@ -3,7 +3,7 @@
 
 class TemperatureModuleContext:  # noqa: D101
     # TODO(mc, 2022-02-09): copy or rewrite docstring from
-    # src/opentrons/protocol_api/module_contexts.py
+    # src/opentrons/legacy/module_contexts.py
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api_experimental/module_contexts/temperature_module_context.py]:
```suggestion
    # src/opentrons/protocol_api/module_contexts.py
```

File: api/src/opentrons/protocol_api_experimental/module_contexts/thermocycler_module_context.py [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 class ThermocyclerModuleContext:  # noqa: D101
     # TODO(mc, 2022-02-09): copy or rewrite docstring from
-    # src/opentrons/protocol_api/module_contexts.py
+    # src/opentrons/legacy/module_contexts.py
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api_experimental/module_contexts/thermocycler_module_context.py]:
```suggestion
    # src/opentrons/protocol_api/module_contexts.py
```

File: api/src/opentrons/protocol_api_experimental/pipette_context.py [hunk 10]
```diff
@@ -12,7 +12,7 @@
 from opentrons.hardware_control.dev_types import PipetteDict
 
 # todo(mm, 2021-04-09): Duplicate these classes in this package to
-# decouple from the v2 opentrons.protocol_api?
+# decouple from the v2 opentrons.legacy?
```
Comment (TamarZanzouri) [inline api/src/opentrons/protocol_api_experimental/pipette_context.py]:
```suggestion
# decouple from the v2 opentrons.protocol_api?
```

File: api/src/opentrons/protocols/duration/estimator.py (hunk 1 above)
Comment (TamarZanzouri) [inline api/src/opentrons/protocols/duration/estimator.py]:
```suggestion
        # protocol_api.labware.Well.top
```

File: api/tests/opentrons/protocol_api/core/engine/__init__.py [hunk 11]
```diff
@@ -1 +1 @@
-"""Tests for opentrons.protocol_api.core.engine."""
+"""Tests for opentrons.legacy.core.engine."""
```
Comment (TamarZanzouri) [inline api/tests/opentrons/protocol_api/core/engine/__init__.py]:
```suggestion
"""Tests for opentrons.protocol_api.core.engine."""
```

File: api/tests/opentrons/protocol_api/core/engine/test_module_core.py [hunk 12]
```diff
@@ -1,4 +1,4 @@
-"""Tests for opentrons.protocol_api.core.engine.ModuleCore."""
+"""Tests for opentrons.legacy.core.engine.ModuleCore."""
```
Comment (TamarZanzouri) [inline api/tests/opentrons/protocol_api/core/engine/test_module_core.py]:
```suggestion
"""Tests for opentrons.protocol_api.core.engine.ModuleCore."""
```

File: api/Pipfile [hunk 13]
```diff
@@ -28,7 +28,7 @@ flake8 = "~=3.9.0"
 flake8-annotations = "~=2.6.2"
 flake8-docstrings = "~=1.6.0"
 flake8-noqa = "~=1.2.1"
-decoy = "~=1.10"
+decoy = "~=1.11.3"
```
Comment (SyntaxColoring) [inline api/Pipfile]: `~=1.11.3` won't do what most people expect, unfortunately. It's equivalent to `>=1.11.3, <1.12`, so it won't allow minor version upgrades. See https://peps.python.org/pep-0440/#compatible-release.

If we need a new Decoy feature only available in newer releases, let's change this to either `>=1.11.3, <2`, or `~=1.11`. You might need to re-lock Pipfile.lock afterwards.

**Comment by TamarZanzouri** [comment]
Smoke tested on Dev server with a bunch of python protocols and they all passed analysis. Need to smoke test on an actual robot.

File: api/src/opentrons/protocol_api/core/legacy_simulator/instrument_context.py [hunk 14]
```diff
@@ -28,7 +28,7 @@
 """In PAPIv2.1 and below, tips are always dropped 10 mm from the bottom of the well."""
 
 
-class InstrumentContextSimulation(AbstractInstrument[WellImplementation]):
+class InstrumentContextSimulation(AbstractInstrument[LegacyWellCore]):
```
Comment (mcous) [inline api/src/opentrons/protocol_api/core/legacy_simulator/instrument_context.py]:
```suggestion
class LegacyInstrumentCoreSimulator(AbstractInstrument[LegacyWellCore]):
```

File: api/src/opentrons/protocol_api/core/legacy_simulator/protocol_context.py [hunk 15]
```diff
@@ -5,21 +5,19 @@
...
+from ..legacy.legacy_labware_core import LegacyLabwareCore
+from ..legacy.legacy_module_core import LegacyModuleCore
+from ..legacy.load_info import InstrumentLoadInfo
 
 from .instrument_context import InstrumentContextSimulation
 
 logger = logging.getLogger(__name__)
 
 
 class ProtocolContextSimulation(
```
Comment (mcous) [inline api/src/opentrons/protocol_api/core/legacy_simulator/protocol_context.py]:
```suggestion
class LegacyProtocolCoreSimulator(
```

**Review by mcous — APPROVED** [review]
This looks great, a couple more renames we can slot in if we want to, but no huge need. @jbleon95 is smoke testing on a robot right now. 

There's one last task before we can close out the ticket, but I think it should be in a separate PR for safety: rename public context init args and properties from `self._impl` / `self._implementation` to `self._core`.

**Comment by TamarZanzouri** [comment]
> This looks great, a couple more renames we can slot in if we want to, but no huge need. @jbleon95 is smoke testing on a robot right now.
> 
> There's one last task before we can close out the ticket, but I think it should be in a separate PR for safety: rename public context init args and properties from `self._impl` / `self._implementation` to `self._core`.

Awesome! I will get this work done in a separate PR and will merge this after @jbleon95 gives his verdict.

**Review by jbleon95 — APPROVED** [review]
Smoke tested and looks good!

**Review by SyntaxColoring — APPROVED** [review]
Thanks!
